msi/release.sh

```diff
@@ -1,42 +1,152 @@
 #!/bin/sh
 
-# Build Git for Windows' .msi files
+# Build Git for Windows .msi file
 
 test -z "$1" && {
-	echo "Usage: $0 <version> [optional components]"
-	exit 1
+    echo
+    echo "Usage: $0 [-v=|--version=|-v ]<version> [(-o=|--outputDir=|-o ) <Output Directory>] [Optional Components]"
+    echo
+    echo "   -v <version>"
+    echo "   -v=<version>"
+    echo "   --version=<version>"
+    echo "       The version of Git for Windows for which to create an installer."
+    echo "       The version should be in the format:"
+    echo
+    echo "           <Major>.<Minor>.<Revision>.<Build/Patch>."
+    echo
+    echo "   -o <Output Directory>"
+    echo "   -o=<Output Directory>"
+    echo "   --outputDir=<OutputDirectory>"
+    echo "       The location the completed installer should be written to."
+    echo "       If this option is not specified, the installer will be written to your"
+    echo "       home directory."
+    echo
+    echo "   [Optional Components]"
+    echo "       A list of additional packages you want bundled with your Git for Windows"
+    echo "       distribution installer."
+    echo
+    echo "       For example, if you want to install emacs with your Git for Windows"
+    echo "       installation, you would run the command:"
+    echo
+    echo "           ./release.sh M.m.r.b mingw-w64-emacs-git"
+    echo
+    echo "       where M.m.r.b is the Major, minor, revision, and build number, "
+    echo "       respectively."
+    echo
+    exit 1
 }
 
 die () {
-	echo "$*" >&1
```

Comment:
It's really hard keeping track of which files are using `{TAB}` indenting vs. `{SPACE}` indenting. I apologize for the whitespace changes, but I tend to "un-tabify" code files. I suppose I should turn on my editor option to view whitespace characters.... Let me know if you'd like me to go back and fix this.